lib/src/source_visitor.dart

```diff
@@ -2289,6 +2289,38 @@ class SourceVisitor extends ThrowingAstVisitor {
     token(node.literal);
   }
 
+  @override
+  void visitObjectPattern(ObjectPattern node) {
+    // Even though object patterns syntactically resemble constructor or
+    // function calls, we format them like collections (or like argument lists
+    // with trailing commas). In other words, like this:
+    //
+    //     case Foo(
+    //         first: 1,
+    //         second: 2,
+    //         third: 3
+    //       ):
+    //       body;
+    //
+    // Not like:
+    //
+    //     case Foo(
+    //           first: 1,
+    //           second: 2,
+    //           third: 3):
+    //       body;
+    //
+    // This is less consistent with the corresponding expression form, but is
+    // more consistent with all of the other delimited patterns -- list, map,
+    // and record -- which have collection-like formatting.
+    // TODO(rnystrom): If we move to consistently using collection-like
```

Comment:
[optional] This is a place where linking the TODO to an issue instead of author will greatly increase the likelihood of it getting noticed along with other changes for that issue.

Reply:
Yeah, that's a good idea. Though I'm hesitant to file an issue for this because of the volume of discussion it will generate that I don't have time to moderate right now.